SEED-platform/seed
# Build/django 3.1

#### Any background context you want to provide?
SEED is updating its django version

#### What's this PR do?
- Update django to 3.1.13
  - Fix any dependencies that were causing issues on this new version
  - Fix non-backwards compatible usage in SEED
- Tweak derived columns testing due to edge case of too deep recursion
- ⚠️ Rename the command for creating geojson data: from `create_geojson_test_data` to `create_geojson_mock_data`. For some reason the test pattern matching for 3.1 was picking up that file which was causing some issues which should probably be investigated separately, see #2895 
- ⚠️  This PR requires a migration to be run -- we accidentally forgot to add a migration previously
- Update our CI workflow to check that all migrations that need to exist do exist...

#### How should this be manually tested?
- If testing locally, make sure your instance is updated (rebuild or pip install) and run the new migration. Try to use many features of SEED and see if anything weird happens

#### logs
For the sake of convenient documentation, here are the logs of the tests when run with the -Wd flag to see all of the nice new Django 4.0 warnings we get:
https://gist.github.com/macintoshpie/707653b2338c9d09db463af3cb273c89
#### What are the relevant tickets?
#2737 

#### Screenshots (if appropriate)
